### PR TITLE
feat: add object search example

### PR DESCRIPTION
**Pull request by bhavsarpratik**

### CODE REVIEW

File: object-search/pods/craft.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+requests:
+  on:
+    IndexRequest:
+      - !CraftDriver
+        with:
+          executor: img_read
+      - !CraftDriver
+        with:
+          executor: img_norm
+      - !CraftDriver
```
Comment (JoanFM) [inline object-search/pods/craft.yml:38]: TorchObjectDetection is a Segmenter, therefore a SegmentDriver should be driving it. I suggest to split these executors in at least 2 steps (load and normalize) and then (detect)

File: object-search/pods/craft.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      channel_axis: -1
+    metas:
+      name: img_norm
+  - !TorchObjectDetectionSegmenter
+    with:
+      channel_axis: -1
+      confidence_threshold: 0.98
+    metas:
+      name: img_detect
+  - !ImageResizer
```
Comment (JoanFM) [inline object-search/pods/craft.yml:22]: Remember that the detected objects will need to be normalized again before encoder step. But at the same time, the documents that need to go to the KVIndexer, do not have to be normalized

File: object-search/pods/vec.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  name: vec_idx  # a customized name
+  workspace: $WORKDIR
+requests:
+  on:
+    ControlRequest:
+      - !ControlReqDriver {}
+    SearchRequest:
+      - !VectorSearchDriver {}
+    IndexRequest:
+      - !ExcludeQL
```
Comment (JoanFM) [inline object-search/pods/vec.yml:14]: no need for this exclusion, since VectorIndexer will get only the embedding anyway

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: object-search/pods/craft.yml [hunk 4]
```diff
@@ -0,0 +1,56 @@
+!CompoundExecutor
+components:
+  - !ImageReader
+    with:
+      from_bytes: True
+    metas:
+      name: img_read
+  - !ImageNormalizer
```
Comment (JoanFM) [inline object-search/pods/craft.yml:8]: Just a note for our improvement, in the `TorchObjectDetectionSegmenter` hub tests, no normalization was applied before, it does not matter but it may lead to a little missuse, because I think since the `backbone` is as any other classification model, it should benefit from this normalization.

File: object-search/flow-index.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+pods:
+  crafter:
+    uses: pods/craft.yml
+    read_only: true
+  encoder:
+    uses: pods/encode.yml
+    parallel: $PARALLEL
+    timeout_ready: 600000
+    read_only: true
+  vec_idx:
```
Comment (JoanFM) [inline object-search/flow-index.yml:13]: I would expect here to have a compound_indexer for `chunk_vec` and `chunk_kv` data, followed by another pod with `doc_kv` index. In this case you will have to be careful how to connect the pods since it is important to know what needs and does not need preprocessing.

File: object-search/pods/craft.yml (hunk 1 above)
Comment (bhavsarpratik) [reply object-search/pods/craft.yml:38]: Hey Joan, I am getting this after I changed `craftdriver` to `segmentdriver`. Can this be due to TorchObjectDetectionSegmenter confidence_threshold: 0.98

```
crafter@37895[W]:doc 1 at level 0 gives no chunk
crafter@37895[W]:doc 2 at level 0 gives no chunk
```

Comment (JoanFM) [reply object-search/pods/craft.yml:38]: Yes, TorchObjectDetection may not detect anything, for testing you can keep a really low threshold.

Comment (bhavsarpratik) [reply object-search/pods/craft.yml:38]: Keeping threshold to 0.1 detects objects in some images but then throws error about different shape of cropped images while passing through the next encoder. I have used resize as shown below.

```
  - !TorchObjectDetectionSegmenter
    with:
      channel_axis: -1
      confidence_threshold: 0.1
    metas:
      name: img_detect
  - !ImageResizer
    with:
      target_size: 96
    metas:
      name: img_resize
```

Also tried

```
  - !TorchObjectDetectionSegmenter
    with:
      target_size: (96, 96)
      channel_axis: -1
      confidence_threshold: 0.1
    metas:
      name: img_detect
  - !ImageResizer
    with:
      target_size: 96
    metas:
      name: img_resize
```

vec_idx@38553[I]:received "index" from gateway▸crafter▸encoder▸⚐
        vec_idx@38553[W]:these bad docs can not be added: [(2450309498, 1), (3283606135, 1)]
        vec_idx@38553[W]:these bad docs can not be added: [(537664483, 4), (1729837641, 4), (3884707008, 4), (1685125731, 4), (1960091019, 4)]
        vec_idx@38553[W]:these bad docs can not be added: [(2830395716, 6), (923790454, 6), (1953433588, 6)]
        vec_idx@38553[W]:these bad docs can not be added: [(1, 0), (2, 0), (3, 0), (4, 0), (5, 0), (6, 0)]
       join_all@38555[I]:received "index" from gateway▸crafter▸encoder▸vec_idx▸⚐
       join_all@38555[I]:collected 2/2 parts of IndexRequest
        gateway@38544[E]:callback() may not work properly due to the bad response: encoder(ImageKerasEncoder) throws ValueError('all input arrays must have the same shape')

Comment (JoanFM) [reply object-search/pods/craft.yml:38]: I think u did not use ImageResizer with the right granularity level, u applied ImageResizer to the Original image and is not what is being indexed. First,, let's divide this crafter in at least 2 different Pods

**Comment by bhavsarpratik** [comment]
@JoanFM Pushed the changes which we finalised. When I run the query it throws this.

      tf_encode@45255[I]:received "search" from gateway▸normalizer▸object▸⚐
      tf_encode@45255[I]:>>>>>>>> EncodeDriver 1 0 []
      tf_encode@45255[E]:data type '' not understood
Traceback (most recent call last):
  File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/peapods/pea.py", line 285, in msg_callback
    self.zmqlet.send_message(self._callback(msg))
  File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/peapods/pea.py", line 259, in _callback
    self.pre_hook(msg).handle(msg).post_hook(msg)
  File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/peapods/pea.py", line 156, in handle
    self.executor(self.request_type)
  File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/executors/__init__.py", line 557, in __call__
    d()
  File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/drivers/__init__.py", line 266, in __call__
    self._traverse_apply(self.req.docs, *args, **kwargs)
  File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/drivers/__init__.py", line 336, in _traverse_apply
    _traverse(working_docs, 'chunks', None, 'granularity', self._granularity_start, self._granularity_end)
  File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/drivers/__init__.py", line 293, in post_traverse
    self._apply_all(_docs, context_doc, recur_on, *args, **kwargs)
  File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/drivers/encode.py", line 31, in _apply_all
    contents, docs_pts, bad_doc_ids = extract_docs(docs, embedding=False)
  File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/drivers/helper.py", line 96, in extract_docs
    content = _extract_fn(doc)
  File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/drivers/helper.py", line 93, in <lambda>
    _extract_fn = lambda doc: doc.text or doc.buffer or (doc.blob and pb2array(doc.blob))
  File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/drivers/helper.py", line 23, in pb2array
    x = np.frombuffer(blob.buffer, dtype=blob.dtype)
TypeError: data type '' not understood
        vec_idx@45260[I]:received "search" from gateway▸normalizer▸object▸tf_encode▸⚐
        vec_idx@45260[E]:list index out of range
Traceback (most recent call last):
  File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/peapods/pea.py", line 285, in msg_callback
    self.zmqlet.send_message(self._callback(msg))
  File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/peapods/pea.py", line 259, in _callback
    self.pre_hook(msg).handle(msg).post_hook(msg)
  File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/peapods/pea.py", line 156, in handle
    self.executor(self.request_type)
  File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/executors/__init__.py", line 557, in __call__
    d()
  File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/drivers/__init__.py", line 266, in __call__
    self._traverse_apply(self.req.docs, *args, **kwargs)
  File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/drivers/__init__.py", line 348, in _traverse_apply
    while working_docs[0].granularity < self._granularity_start:
IndexError: list index out of range
        doc_idx@45261[I]:received "search" from gateway▸normalizer▸object▸tf_encode▸vec_idx▸⚐
         ranker@45262[I]:received "search" from gateway▸normalizer▸object▸tf_encode▸vec_idx▸doc_idx▸⚐
        gateway@45263[I]:#sent: 1 #recv: 1 sent_size: 185.0 KB recv_size: 375.5 KB

**Comment by JoanFM** [comment]
> @JoanFM Pushed the changes which we finalised. When I run the query it throws this.
> 
> tf_encode@45255[I]:received "search" from gateway▸normalizer▸object▸⚐
> tf_encode@45255[I]:>>>>>>>> EncodeDriver 1 0 []
> tf_encode@45255[E]:data type '' not understood
> Traceback (most recent call last):
> File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/peapods/pea.py", line 285, in msg_callback
> self.zmqlet.send_message(self._callback(msg))
> File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/peapods/pea.py", line 259, in _callback
> self.pre_hook(msg).handle(msg).post_hook(msg)
> File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/peapods/pea.py", line 156, in handle
> self.executor(self.request_type)
> File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/executors/**init**.py", line 557, in **call**
> d()
> File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/drivers/**init**.py", line 266, in **call**
> self._traverse_apply(self.req.docs, *args, **kwargs)
> File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/drivers/**init**.py", line 336, in _traverse_apply
> _traverse(working_docs, 'chunks', None, 'granularity', self._granularity_start, self._granularity_end)
> File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/drivers/**init**.py", line 293, in post_traverse
> self._apply_all(_docs, context_doc, recur_on, *args, **kwargs)
> File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/drivers/encode.py", line 31, in _apply_all
> contents, docs_pts, bad_doc_ids = extract_docs(docs, embedding=False)
> File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/drivers/helper.py", line 96, in extract_docs
> content = _extract_fn(doc)
> File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/drivers/helper.py", line 93, in
> _extract_fn = lambda doc: doc.text or doc.buffer or (doc.blob and pb2array(doc.blob))
> File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/drivers/helper.py", line 23, in pb2array
> x = np.frombuffer(blob.buffer, dtype=blob.dtype)
> TypeError: data type '' not understood
> vec_idx@45260[I]:received "search" from gateway▸normalizer▸object▸tf_encode▸⚐
> vec_idx@45260[E]:list index out of range
> Traceback (most recent call last):
> File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/peapods/pea.py", line 285, in msg_callback
> self.zmqlet.send_message(self._callback(msg))
> File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/peapods/pea.py", line 259, in _callback
> self.pre_hook(msg).handle(msg).post_hook(msg)
> File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/peapods/pea.py", line 156, in handle
> self.executor(self.request_type)
> File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/executors/**init**.py", line 557, in **call**
> d()
> File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/drivers/**init**.py", line 266, in **call**
> self._traverse_apply(self.req.docs, *args, **kwargs)
> File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/drivers/**init**.py", line 348, in _traverse_apply
> while working_docs[0].granularity < self._granularity_start:
> IndexError: list index out of range
> doc_idx@45261[I]:received "search" from gateway▸normalizer▸object▸tf_encode▸vec_idx▸⚐
> ranker@45262[I]:received "search" from gateway▸normalizer▸object▸tf_encode▸vec_idx▸doc_idx▸⚐
> gateway@45263[I]:#sent: 1 #recv: 1 sent_size: 185.0 KB recv_size: 375.5 KB

Can u do some cleaning of the old files and the new ones so that it is easier to see what's happening?

**Comment by bhavsarpratik** [comment]
Currently, there is only craft-old which is not used. I will push total cleanup in my next one.

**Comment by JoanFM** [comment]
> > @JoanFM Pushed the changes which we finalised. When I run the query it throws this.
> > tf_encode@45255[I]:received "search" from gateway▸normalizer▸object▸⚐
> > tf_encode@45255[I]:>>>>>>>> EncodeDriver 1 0 []
> > tf_encode@45255[E]:data type '' not understood
> > Traceback (most recent call last):
> > File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/peapods/pea.py", line 285, in msg_callback
> > self.zmqlet.send_message(self._callback(msg))
> > File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/peapods/pea.py", line 259, in _callback
> > self.pre_hook(msg).handle(msg).post_hook(msg)
> > File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/peapods/pea.py", line 156, in handle
> > self.executor(self.request_type)
> > File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/executors/**init**.py", line 557, in **call**
> > d()
> > File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/drivers/**init**.py", line 266, in **call**
> > self._traverse_apply(self.req.docs, *args, **kwargs)
> > File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/drivers/**init**.py", line 336, in _traverse_apply
> > _traverse(working_docs, 'chunks', None, 'granularity', self._granularity_start, self._granularity_end)
> > File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/drivers/**init**.py", line 293, in post_traverse
> > self._apply_all(_docs, context_doc, recur_on, *args, **kwargs)
> > File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/drivers/encode.py", line 31, in _apply_all
> > contents, docs_pts, bad_doc_ids = extract_docs(docs, embedding=False)
> > File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/drivers/helper.py", line 96, in extract_docs
> > content = _extract_fn(doc)
> > File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/drivers/helper.py", line 93, in
> > _extract_fn = lambda doc: doc.text or doc.buffer or (doc.blob and pb2array(doc.blob))
> > File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/drivers/helper.py", line 23, in pb2array
> > x = np.frombuffer(blob.buffer, dtype=blob.dtype)
> > TypeError: data type '' not understood
> > vec_idx@45260[I]:received "search" from gateway▸normalizer▸object▸tf_encode▸⚐
> > vec_idx@45260[E]:list index out of range
> > Traceback (most recent call last):
> > File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/peapods/pea.py", line 285, in msg_callback
> > self.zmqlet.send_message(self._callback(msg))
> > File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/peapods/pea.py", line 259, in _callback
> > self.pre_hook(msg).handle(msg).post_hook(msg)
> > File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/peapods/pea.py", line 156, in handle
> > self.executor(self.request_type)
> > File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/executors/**init**.py", line 557, in **call**
> > d()
> > File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/drivers/**init**.py", line 266, in **call**
> > self._traverse_apply(self.req.docs, *args, **kwargs)
> > File "/opt/miniconda3/envs/jina/lib/python3.7/site-packages/jina/drivers/**init**.py", line 348, in _traverse_apply
> > while working_docs[0].granularity < self._granularity_start:
> > IndexError: list index out of range
> > doc_idx@45261[I]:received "search" from gateway▸normalizer▸object▸tf_encode▸vec_idx▸⚐
> > ranker@45262[I]:received "search" from gateway▸normalizer▸object▸tf_encode▸vec_idx▸doc_idx▸⚐
> > gateway@45263[I]:#sent: 1 #recv: 1 sent_size: 185.0 KB recv_size: 375.5 KB
> 
> Can u do some cleaning of the old files and the new ones so that it is easier to see what's happening?

The encodeDriver is getting `doc.id 1` which should not be because it should receive chunks!

File: object-search/pods/vec.yml (hunk 3 above)
Comment (bhavsarpratik) [reply object-search/pods/vec.yml]: This file was removed in favor of the compound indexer

File: object-search/pods/vec.yml [hunk 6]
```diff
@@ -18,6 +18,7 @@ metas:
 requests:
   on:
     IndexRequest:
+      # - !Buffer2URI {}
```
Comment (JoanFM) [inline object-search/pods/vec.yml:21]: Ok try putting between `!VectorIndexDriver` and` !KVIndexDriver` a `!Blob2PngURI` driver. I Think that is the buggest difference with cxmodal search, is that at this point, u are indexing the ouptut of a segmenter that provides ''blob'' and not 'buffer' directly in output

File: object-search/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,12 @@
+FROM jinaai/jina:0.5.4
```
Comment (nan-wang) [inline object-search/Dockerfile:1]: Please make sure this is consistent with the `requirements.txt`

File: fashion-example-query/fashion/app.py [hunk 8]
```diff
@@ -162,21 +162,22 @@ def config():
...
         'index': {
-            'url': 'http://fashion-mnist.s3-website.eu-central-1.amazonaws.com/train-images-idx3-ubyte.gz',
+            'url': 'https://github.com/zalandoresearch/fashion-mnist/raw/master/data/fashion/train-images-idx3-ubyte.gz',
             'filename': os.path.join('./workspace', 'index')
         },
         'query': {
-            'url': 'http://fashion-mnist.s3-website.eu-central-1.amazonaws.com/t10k-images-idx3-ubyte.gz',
+            'url': 'https://github.com/zalandoresearch/fashion-mnist/raw/master/data/fashion/t10k-images-idx3-ubyte.gz',
             'filename': os.path.join('./workspace', 'query')
         }
```
Comment (nan-wang) [inline fashion-example-query/fashion/app.py:182]: it is better to do this in another PR.

File: object-search/.dockerignore [hunk 9]
```diff
@@ -0,0 +1,14 @@
+# ignore everything
+#data
+pretrained
+workspace
+*.html
+
```
Comment (nan-wang) [inline object-search/.dockerignore:6]: maybe add `data` as well?

**Comment by nan-wang** [comment]
building the docker image by `docker build -t jinaai/hub.app.objectsearch .` leads to the following error. Please make sure the commands in the `README.md` are fully tested.

```bash
Error: Got unexpected extra argument (dryrun)
The command '/bin/sh -c python app.py dryrun' returned a non-zero code: 2
```

File: object-search/README.md [hunk 10]
```diff
@@ -0,0 +1,253 @@
...
+<a href="https://twitter.com/intent/tweet?text=%F0%9F%91%8DCheck+out+Jina%3A+the+New+Open-Source+Solution+for+Neural+Information+Retrieval+%F0%9F%94%8D%40JinaAI_&url=https%3A%2F%2Fgithub.com%2Fjina-ai%2Fjina&hashtags=JinaSearch&original_referer=http%3A%2F%2Fgithub.com%2F&tw_p=tweetbutton" target="_blank">
+  <img src="https://github.com/jina-ai/jina/blob/master/.github/badges/twitter-badge.svg"
+       alt="tweet button" title="👍Share Jina with your friends on Twitter"></img>
+</a>
+[![Python 3.7 3.8](https://github.com/jina-ai/jina/blob/master/.github/badges/python-badge.svg "Jina supports Python 3.7 and above")](#)
+[![Docker](https://github.com/jina-ai/jina/blob/master/.github/badges/docker-badge.svg "Jina is multi-arch ready, can run on differnt architectures")](https://hub.docker.com/r/jinaai/jina/tags)
+
+</p>
+
+In this example, we use `fasterrcnn_resnet50_fpn` for object detection  and then index them with `MobileNetV2`. You can use this demo system to index an image dataset and query the most similar object from it.
```
Comment (JoanFM) [inline object-search/README.md:20]: It could be left more clear that you index the detected objects as single images extracted from the original ones.

File: object-search/README.md [hunk 11]
```diff
@@ -0,0 +1,253 @@
...
+
+To save you from dependency hell, we'll use the containerized version in these instructions. That means you only need to have [Docker installed](https://docs.docker.com/get-docker/). No Python virtualenv, no Python package (un)install. 
+
+The code can of course run natively on your local machine, please [read the Jina installation guide for details](https://docs.jina.ai/chapters/install/via-pip.html).
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [TL;DR: Just Show Me the Pokemon!](#tldr-just-show-me-the-pokemon)
```
Comment (JoanFM) [inline object-search/README.md:41]: Remember to change this from Pokemon

File: object-search/README.md [hunk 12]
```diff
@@ -0,0 +1,253 @@
...
+
+with f:
+    f.index(input_fn, batch_size=128)
+```
+
+</td>
+<td>
+  <sub>
+
+```yaml
```
Comment (JoanFM) [inline object-search/README.md:115]: Either make sure to align this with the actual Flows, or remove it and have a smaller README.md

File: object-search/README.md [hunk 13]
```diff
@@ -0,0 +1,253 @@
...
+If it's running successfully, you should be able to see logs scrolling in the console and in the dashboard:
+
+<p align="center">
+  <img src=".github/.README_images/0a8863abb3fcee182e1fe8fe46c47b7a.gif?raw=true" alt="Jina banner" width="45%">
+  <img src=".github/.README_images/ed2907cd11ac26a2a3a2555f16071d13.gif?raw=true" alt="Jina banner" width="45%">
+</p>
+
+Under `$(pwd)/workspace`, you'll see a list of directories `chunk_compound_indexer-*` after indexing. This is because we set shards to 8.
+
+## Query Top-K Visually Similar Images
```
Comment (JoanFM) [inline object-search/README.md:162]: Similar Object Images?

File: object-search/README.md [hunk 14]
```diff
@@ -0,0 +1,253 @@
...
+```
+
+[JSON payload syntax and spec can be found in the docs](https://docs.jina.ai/chapters/restapi/#).
+
+This example shows you how to feed data into Jina via REST gateway. By default, Jina uses a gRPC gateway, which has much higher performance and rich features. If you are interested in that, go ahead and check out our [other examples](https://learn.jina.ai) and [read our documentation on Jina IO](https://docs.jina.ai/chapters/io/#).
+
+### Query Results in Batch
+
+Let's test the results on Pokémon! This time we use our gRPC gateway (for better efficiency in batch querying): Simply run `python make_html.py`
```
Comment (JoanFM) [inline object-search/README.md:187]: Remove this part, since we force the user to use the `dashboard`

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: object-search/pods/vec.yml [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+          traversal_paths: ['c']
+    SearchRequest:
+      - !ExcludeQL
+        with:
+          traversal_paths: ['r']
+          fields:
+            - uri  
+            - buffer
+            - blob
+      - !LogInfoDriver {}
```
Comment (nan-wang) [inline object-search/pods/vec.yml:38]: `LogInfoDriver` should be cleaned up.

File: object-search/flow-index.yml [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+with:
+  logserver: false
+pods:
+  normalizer:
+    uses: pods/craft-normalize.yml
+    read_only: true
+  object:
+    uses: pods/craft-object.yml
+    read_only: true
+  resize:
```
Comment (JoanFM) [inline object-search/flow-index.yml:11]: ok, complex situation, to keep quality of the image indexed.
Right now the `chunk.gz` contains images resized with a loss of quality because they need to be resized to be encoded. 

Steps:

1. Need to split vec.yml into a `NumpyIndexer` and `BinaryPbIndexer` in different Pods.
2. The new BinaryPbIndexer needs to receive directly the output from `craft-object` Pod.

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: object-search/README.md [hunk 17]
```diff
@@ -17,9 +17,7 @@
 
 </p>
 
-In this example, we use `fasterrcnn_resnet50_fpn` for object detection  and then index them with `MobileNetV2`. You can use this demo system to index an image dataset and query the most similar object from it. 
-
-[![](.github/.README_images/7262e2aa.png)](https://get.jina.ai)
+In this example, we use `fasterrcnn_resnet50_fpn` for object detection and then index these cropped object images with `MobileNetV2`. You can use this demo system to index an image dataset and query the most similar object in those images.
```
Comment (JoanFM) [inline object-search/README.md:20]: I would emphasize that the expectrd input images for query are objects image and not general images since the indexed part is the cropped detrected image

File: object-search/README.md [hunk 18]
```diff
@@ -38,37 +36,23 @@ The code can of course run natively on your local machine, please [read the Jina
...
-
-```bash
-docker run -p 34567:34567 -e "JINA_PORT=34567" jinaai/hub.app.bitsearch-pokedex search
-```
-
-Then you can `curl`/query/js it via HTTP POST request. [Details here](#query-via-rest-api).
-
 ## Download and Extract Data
 ### Use Flickr8k
 Although the model is trained on Flickr30k, you can test on Flickr8k dataset, which is a much smaller version of
```
Comment (JoanFM) [inline object-search/README.md:52]: It is not valid that the model has been ttained in f30k, we should emphasize that it is not the best dataset for the job and this kind of stuff. That they are entitled to use any other dataset

File: object-search/README.md [hunk 19]
```diff
@@ -78,149 +62,63 @@ mv Images data/f8k/images
 ```
 
 ## Index Image Data
-
+Index 16 images
```
Comment (JoanFM) [inline object-search/README.md:65]: Why 16?

File: object-search/README.md [hunk 20]
```diff
@@ -78,149 +62,63 @@ mv Images data/f8k/images
...
+```
 
-## Query Top-K Visually Similar Images
+<p align="center">
+  <img src=".github/.README_images/dog.png?raw=true" alt="Jina banner" width="45%">
+  <img src=".github/.README_images/dog-results.png?raw=true" alt="Jina banner" width="45%">
+</p>
 
-### Start the Jina server
+Start server which returns object images
```
Comment (JoanFM) [inline object-search/README.md:88]: Try to emphasize the difference between these 2 modes

**Review by JoanFM — COMMENTED** [review]
At the end we could have @alexcg1 review the Readme

**Comment by alexcg1** [comment]
For sure!

File: object-search/README.md (hunk 19 above)
Comment (bhavsarpratik) [reply object-search/README.md:65]: Because it takes a lot of time to index, I thought of mentioning a small number for people to try it out.

Comment (JoanFM) [reply object-search/README.md:65]: I woukd have it generic with a recommendation to start with a small number to test.

File: object-search/README.md [hunk 21]
```diff
@@ -49,22 +49,21 @@ The code can of course run natively on your local machine, please [read the Jina
 
 ## Download and Extract Data
 ### Use Flickr8k
-Although the model is trained on Flickr30k, you can test on Flickr8k dataset, which is a much smaller version of 
-flickr30k.
+You can try this example with [Flickr8k](https://www.kaggle.com/adityajn105/flickr8k) object detection dataset. You can also use other datasets like COCO & [Open Images 2019](https://www.kaggle.com/c/open-images-2019-object-detection/overview)
```
Comment (JoanFM) [inline object-search/README.md:52]: A warning that is not ideal should be made?

File: object-search/README.md [hunk 22]
```diff
@@ -49,22 +49,21 @@ The code can of course run natively on your local machine, please [read the Jina
...
 
-To do so, instead of downloading the Flickr8k from Kaggle
+Download the Flickr8k from Kaggle
 
 ```bash
 kaggle datasets download adityajn105/flickr8k
 unzip flickr8k.zip 
 rm flickr8k.zip
-mv Images data/f8k/images
+mv Images data/flickr8k/images
```
Comment (JoanFM) [inline object-search/README.md:60]: Can it be left to f8k so that is coherent with cxmodal search?

**Review by JoanFM — APPROVED** [review]
